tmt/result.py

```diff
@@ -302,31 +322,54 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret in (ResultInterpret.RESPECT, ResultInterpret.CUSTOM):
+        if interpret == ResultInterpret.CUSTOM:
+            return self
+
+        # Check for failed checks
+        checks_failed = any(check.result == ResultOutcome.FAIL for check in self.check)
+
+        if interpret == ResultInterpret.RESPECT:
+            if self.respect_checks and checks_failed:
+                self.result = ResultOutcome.FAIL
+                if self.note:
+                    self.note += ', check failed'
+                else:
+                    self.note = 'check failed'
```

Comment:
@psss @thrix @lukaszachy what do you think about changing results’ `note` field to a list of strings? It is clearly used like a list, various places add their notes, they always have to chech whether there's any note already, add a comma… If it looks like a list of notes, and it quacks like a list of notes…

Will it break anything, e.g. in TF?